Dockerfile

```diff
@@ -90,48 +79,80 @@ ADD . ${BRAYNS_SRC}
 
 # Install Brayns
 # https://github.com/BlueBrain/Brayns
-RUN cksum ${BRAYNS_SRC}/.gitsubprojects && \
-    cd ${BRAYNS_SRC} && \
-    apt-get update && \
-    apt-get -y install ${BUILD_TOOLS} && \
-    apt-get -y install libtbb-dev \
+RUN cksum ${BRAYNS_SRC}/.gitsubprojects \
+ && cd ${BRAYNS_SRC} \
+ && apt-get -y install \
     freeglut3-dev \
     libassimp-dev \
     libboost-all-dev \
+    libglew-dev \
     libglu1-mesa-dev \
     libhdf5-serial-dev \
     libjpeg-turbo8-dev \
     libmagick++-dev \
+    libtbb-dev \
     libturbojpeg \
     libxi-dev \
-    libzmq3-dev \
     libxmu-dev \
+    libzmq3-dev \
     python-pyparsing \
     qtbase5-dev \
     qtdeclarative5-dev \
-    xorg-dev && \
-    git submodule update --init --recursive --remote && \
-    mkdir -p build && \
-    cd build && \
-    cmake .. -GNinja \
-    -DCMAKE_INSTALL_PREFIX=${DIST_PATH} \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DBRAYNS_NETWORKING_ENABLED=ON \
-    -DBRAYNS_DEFLECT_ENABLED=ON \
+    xorg-dev \
+ && git submodule update --init --recursive --remote \
+ && mkdir -p build \
+ && cd build \
+ && cmake .. -GNinja \
     -DBRAYNS_BRION_ENABLED=ON \
-    -DCLONE_SUBPROJECTS=ON && \
-    ninja install && \
-    cd /app && \
-    rm -rf ${BRAYNS_SRC}/build && \
-    apt-get -y remove ${BUILD_TOOLS} && \
-    apt-get -y purge *-dev python-pyparsing && \
-    apt-get clean
+    -DBRAYNS_DEFLECT_ENABLED=ON \
+    -DBRAYNS_NETWORKING_ENABLED=ON \
+    -DCLONE_SUBPROJECTS=ON \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=${DIST_PATH} \
+ && ninja install
 
+# Final image, containing only Brayns and libraries required to run it
+FROM ubuntu:xenial
+RUN apt-get update \
+ && apt-get install -y \
+    freeglut3 \
+    libassimp3v5 \
+    libboost-atomic1.58.0 \
+    libboost-chrono1.58.0 \
+    libboost-date-time1.58.0 \
+    libboost-filesystem1.58.0 \
+    libboost-program-options1.58.0 \
+    libboost-regex1.58.0 \
+    libboost-serialization1.58.0 \
+    libboost-system1.58.0 \
+    libboost-thread1.58.0 \
+    libglew1.13 \
+    libgomp1 \
+    libhdf5-10 \
+    libhdf5-cpp-11 \
+    libhdf5-cpp-11 \
+    libhwloc5 \
+    libmagick++-6.q16-5v5 \
+    libmagickwand-6.q16-2 \
+    libqt5concurrent5 \
+    libqt5network5 \
+    libtbb2 \
+    libturbojpeg \
+    libxmu6 \
+    libzmq5 \
+ && rm -rf /var/lib/apt/lists/*
+
+# The COPY command below will:
+# 1. create a container based on the `builder` image (but do not start it)
+#    Equivalent to the `docker create` command
+# 2. create a new image layer containing the
+#    /app/dist directory of this new container
+#    Equivalent to the `docker copy` command.
+COPY --from=builder /app/dist /app/dist
 
 # Add binaries from dist to the PATH
-ENV LD_LIBRARY_PATH $LD_LIBRARY_PATH:${DIST_PATH}/lib
-ENV PATH $PATH:${DIST_PATH}/bin
-
+ENV LD_LIBRARY_PATH $LD_LIBRARY_PATH:/app/dist/lib
```

Comment:
why not reusing DIST_PATH instead of hard-coding /app/dist/lib?

Reply:
I agree, just use `DIST_PATH`.

Reply:
1. You are right, it should not be hardcoded. I will make the required changes.
1. Besides `ENV` and `ARG` are not available in other build stages. they must be redefined.
1. I think we should use `ARG` instead of `ENV` as `DIST_PATH` does not seem to be a required variable in the container. Anybody against?